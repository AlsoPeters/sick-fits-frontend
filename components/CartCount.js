export default function CartCount({ count }) {
  return (
    <div className="p-2 mx-2 text-base font-bold leading-4 rounded-full tabular-nums h-min min-w-fit bg-tokyo-term-black">
      {count}
    </div>
  );
}
