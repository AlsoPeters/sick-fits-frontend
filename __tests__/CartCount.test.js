import { render } from "@testing-library/react";
import wait from "waait";
import CartCount from "../components/CartCount";

describe("<CartCount/>", () => {
  it("Renders", () => {
    render(<CartCount count={10} />);
  });
  it("Matches snapshot", () => {
    const { container } = render(<CartCount count={11} />);
    expect(container).toMatchSnapshot();
  });
  it("Updates via props", () => {
    const { container, rerender, debug } = render(<CartCount count={11} />);
    expect(container.textContent).toBe("11");
    // expect(container).toHaveTextContent('11');
    // Update the props
    rerender(<CartCount count={12} />);
    debug();
    expect(container.textContent).toBe("12");
  });
});
