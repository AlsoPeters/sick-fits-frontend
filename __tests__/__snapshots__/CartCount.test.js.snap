// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartCount/> Matches snapshot 1`] = `
<div>
  <div
    class="p-2 mx-2 text-base font-bold leading-4 rounded-full tabular-nums h-min min-w-fit bg-tokyo-term-black"
  >
    11
  </div>
</div>
`;
